wp/resources/sample_binaries/pointer_flag/run_wp_sat.sh

```diff
@@ -0,0 +1,23 @@
+# This is a test that tests that the wp-pointer-reg-list flag works as expected.
+# Specifically, the function foo tries to use differing
+# return addresses that are pushed onto the stack to cause a different return
+# value between functions. The abscence of the flag should allow the pointer
+# argument in RDI to be used to point to this return address, as it lives
+# on the uninitialized portion of the stack at the start of the function.
+
+# Should return SAT
```

Comment:
I actually don't get how this example is supposed to work. What's happening with `num1`? It might help to actually refer to the C names of variables and functions.